pkg/gofr/datasource/surrealdb/surrealDB.go

```diff
@@ -0,0 +1,335 @@
+package surrealdb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	surrealdb "github.com/surrealdb/surrealdb.go"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+
+	"time"
+)
+
+var (
+	// ErrNotConnected indicates that the database client is not connected.
+	ErrNotConnected = errors.New("not connected to database")
+)
+
+// Config represents the configuration required to connect to SurrealDB.
+type Config struct {
+	Host       string
+	Port       int
+	Username   string
+	Password   string
+	Namespace  string
+	Database   string
+	TLSEnabled bool
+}
+
+// Client represents a client to interact with SurrealDB.
+type Client struct {
+	config  Config
+	db      *surrealdb.DB
+	logger  Logger
+	metrics Metrics
+	tracer  trace.Tracer
+}
+
+// New creates a new Client with the provided configuration.
+func New(config Config) *Client {
+	return &Client{
+		config: config,
+	}
+}
+
+// UseLogger sets a custom logger for the Client.
+func (c *Client) UseLogger(logger interface{}) {
+	if l, ok := logger.(Logger); ok {
+		c.logger = l
+	}
+}
+
+// UseMetrics sets a custom metrics recorder for the Client.
+func (c *Client) UseMetrics(metrics interface{}) {
+	if m, ok := metrics.(Metrics); ok {
+		c.metrics = m
+	}
+}
+
+// UseTracer sets a custom tracer for the Client.
+func (c *Client) UseTracer(tracer interface{}) {
+	if t, ok := tracer.(trace.Tracer); ok {
+		c.tracer = t
+	}
+}
+
+// Connect establishes a connection to the SurrealDB server using the client's configuration.
+func (c *Client) Connect() {
```

Comment:
For me, this method should return an error, and the error currently logged should be wrapped and returned.

We should be able to stop everything if Connect fails